src/db.rs

```diff
@@ -594,7 +594,7 @@ impl DbInner {
 	}
 
 	fn cleanup_logs(&self) -> Result<bool> {
-		let keep_logs = if self.options.sync_data { KEEP_LOGS } else { 0 };
+		let keep_logs = if self.options.sync_data { 0 } else { KEEP_LOGS };
```

Comment:
This was a bug. The values should be reversed. We only need to keep `KEEP_LOGS` old files when sync is disabled. This gives at least some chance to recover after the power failure or OS crash.